velox/functions/lib/window/NthValueBase.h

```diff
@@ -96,6 +70,18 @@ class NthValueFunction : public exec::WindowFunction {
     partitionOffset_ += numRows;
   }
 
+ protected:
+  // This is needed to retrieve column values from the partition data.
+  column_index_t offsetIndex_;
+
+  // These fields are set if the offset argument is a constant value.
+  std::optional<vector_size_t> constantOffset_;
```

Comment:
I find it a bit anti-pattern that these variables are made protected and set in the child constructor the way they are.

Its cleaner and safer to make the variables private, make them parameters in the constructor and have the child class send the parameters as required from its constructor.

Or add a protected set function for these variables in the base class and call it from the child constructor.

Reply:
Thanks for your suggestion, updated.